### DISCUSSION TITLE
Auto scaling of solar mppts

### DISCUSSION THREAD

**Discussion by lamixer** [slipx06/sunsynk-power-flow-card#321] [General]
Hello:

I have auto-scaling on for solar, and have three mppts configured. The scaling works for the lowest box that shows the sum of the mppts (each string), but the individual ones are always in kW. So I might see 0.05 kW for each mppt but see 150 W for the sum. I'd like the individual ones to switch to watts when below 1 kW.

**Comment by lamixer** [comment]
that part of my config...

```solar:
  show_daily: true
  mppts: 3
  auto_scale: true
  max_power: 7000
  pv1_name: 1 South F
  pv2_name: 2 South R
  pv3_name: 3 North
```

**Reply by slipx06** [reply]
Hi. That's what it's supposed to be doing but there must be an issue interpreting the provided the sensor data. Can you share your sensor details. Are your values W of kW?

![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/7227275/702891a0-cc2b-4e1c-b213-349f26d61462)

**Reply by lamixer** [reply]
I'm in kW I guess...

`sensor.foxess_pv1_power`

`0.423`

```
state_class: measurement
unit_of_measurement: kW
device_class: power
icon: mdi:solar-power-variant-outline
friendly_name: PV1 Power (FoxESS)
```

**Reply by slipx06** [reply]
OK. I can modify the code to catch this

**Reply by lamixer** [reply]
Great... thanks!

**Reply by slipx06** [reply]
give https://github.com/slipx06/sunsynk-power-flow-card/releases/tag/v4.24.5 a go

**Reply by lamixer** [reply]
They are all under 1kW just now and are showing watts. Assuming they scale back up to kW when appropriate (I'll let you know if they don't), that fixed it. Thanks!

**Reply by lamixer** [reply]
(sun came back -- is scaling to kW)
